src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -73,6 +74,97 @@ get_pcidev(const xrt_core::device* device)
   return pcidev::get_dev(device->get_device_id(), device->is_userpf());
 }
 
+static query::sdm_sensor_info::result_type
+get_sdm_sensors(const xrt_core::device* device,
+                const query::sdr_req_type req_type,
+                const std::string path)
+{
+    auto pdev = get_pcidev(device);
+    query::sdm_sensor_info::result_type output;
+    query::sdm_sensor_info::data_type data;
+    std::vector<int> target_sysfs;
+    //All device sensors are stored in hwmon sysfs nodes with starts & ends as follows.
+    std::array<std::string, 5> start = {"curr", "in", "power", "temp", "fan"};
+    std::array<std::string, 5> end = {"label", "input", "max", "average", "highest"};
+    int max_end_types = end.size();
+
+    switch(req_type) {
+      case query::sdr_req_type::current:
+      case query::sdr_req_type::voltage:
+      case query::sdr_req_type::power:
+      case query::sdr_req_type::thermal: //temperature readings
+      case query::sdr_req_type::mechanical: //fan readings
+        target_sysfs.push_back((int)req_type);
+        break;
+      case query::sdr_req_type::electrical:
+        //electrical combines curr + volt + power readings
+        target_sysfs.push_back(0);
+        target_sysfs.push_back(1);
+        target_sysfs.push_back(2);
+        break;
+    }
+
+    //Iterate over hwmon sysfs nodes based on sensor request type.
+    for (const auto& sid : target_sysfs) {
+      bool next_id = false;
+      //All sensor sysfs nodes starts with 1 as starting index.
+      int start_id = 1;
+      int end_id = 0;
+      std::string type = start[sid];
+
+      //voltage sensor sysfs node starts with "in" with 0 as starting index.
+      if (!type.compare("in"))
+        start_id = 0;
+
+      while (!next_id) {
+        end_id = 0;
+        data.label = "N/A";
+        data.input = 0;
+        data.max = 0;
+        data.average = 0;
+        data.highest = 0;
+        /*
+         * Forming sysfs node as /<start>[start_id]_.
+         * Example: /in0_ or /curr1_ or /power1_ or /temp1_ in 1st iteration.
+         * start_id will be incremented till next_id become true.
+         * So, next iteration will be /in1_ or /curr2_ or /power2_ or /temp2_.
+         * Similarly, end string of sysfs node name will be retrieved using end[].
+         */
+        const auto slash = "/";
+        const auto underscore = "_";
+        std::string tmp = slash + type + std::to_string(start_id) + underscore;
+        while (end_id < max_end_types) {
+          if (end_id == 0) {
+            std::string errmsg;
+            std::string label;
+	    pdev->sysfs_get("", path + tmp + end[end_id], errmsg, label);
+	    if (errmsg.empty()) {
+                  data.label = label;
+	    } else {
+		  //go and read next sysfs node
+                  next_id = true;
+		  end_id = max_end_types;
+	    }
+          } else {
```

Comment:
Use early-exit or refactor, I cannot comprehend these many levels.
Please also use the bracing style we embrace (pun intended):

```
if (cond1)
  stmt1;

if (cond2) {
  stmt1;
  stmt2;
}
```

And use proper indentation following what the file is using.

Reply:
I fixed indentation. I didn't get other suggestion related to bracing style. Copying the if else code below. 
Do I need to remove brances for if condition since it has only 1 line of code? 

if (errmsg.empty()) {
   data.label = label;
} else {
  //go and read next sysfs node
  next_id = true;
  end_id = max_end_types;
}

Reply:
It's fine, but `else` should line up with `if`.

Reply:
ok, so the code will look like below. Shall I change to below style? 

```
          if (errmsg.empty()) {
            data.label = label;
          }
          else {
            //go and read next sysfs node
            next_id = true;
            end_id = max_end_types;
          }
```